pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -292,6 +293,18 @@ func buildHTTPVirtualServices(
 			vs.Route = route
 		}
 
+		if r.Timeouts != nil {
+			if r.Timeouts.Request != nil {
+				vs.Timeout = durationpb.New(r.Timeouts.Request.Duration)
+			}
+			if r.Timeouts.BackendRequest != nil {
+				if vs.Retries == nil {
+					vs.Retries = &istio.HTTPRetry{Attempts: 1}
```

Comment:
This is not quite right. Setting a BackendRequest timeout will have an undesirable side effect of doing 1 retry. The problem is that Istio complains if the PerTryTimeout is set when Attempts is 0: `configuration is invalid: http retry policy configured when attempts are set to 0 (disabled)`.

@howardjohn Do you think we can change the validation to allow PerTryTimeout with Attempts=0, or will Envoy also not support that?

I suppose we could handle this by setting vs.Timeout to the value of BackendTimeout in the case that vs.Retries == nil?

Reply:
> I suppose we could handle this by setting vs.Timeout to the value of BackendTimeout in the case that vs.Retries == nil?

:+1: 

BTW, do we validate BackendTimeout >= PerTryTimeout?

Reply:
> BTW, do we validate BackendTimeout >= PerTryTimeout?

Not sure what you are asking?

The gateway-api validator does check that timeouts.BackendRequest <= timeouts.Request, which means that vs.Retries.PerTryTimout will always be <= vs.Timeout.

Reply:
Oh, I see now, that is what you were asking, so the answer is yes 😃